### ISSUE TITLE
Loader is in someway changing the style

### ISSUE THREAD

**Issue by pedroven** [wrabit/django-cotton#55]
Hello,

Firstly, thank you for your project it's exactly what I've been searching for for quite some time.

I'm currently testing django-cotton, and one issue I've noticed is that the Loader seems to be affecting some application styles in a certain way.

```html
<nav class="-mb-px flex space-x-8" aria-label="Tabs">
  <a hx-get="{% url 'route_name' pk=obj.pk %}?search={{ params.items|get_filter_parameters_excluding }}&page=1&order={{ order }}&tab=test1&page_size={{ page_size }}"
      hx-swap="outerHTML"
      hx-target="#obj-page"
      hx-push-url="true"
      hx-indicator=".transitioned-on-loading"
      {% if tab == "test1" %} class="cursor-pointer border-indigo-500 text-indigo-600 dark:text-indigo-500 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" aria-current="page" {% else %} class="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" {% endif %}>
    <span class="text-center sm:text-left">Test 1</span>
  </a>
  <a hx-get="{% url 'route_name' pk=obj.pk %}?search={{ params.items|get_filter_parameters_excluding }}&page=1&order={{ order }}&tab=test2&page_size={{ page_size }}"
      hx-swap="outerHTML"
      hx-target="#obj-page"
      hx-push-url="true"
      hx-indicator=".transitioned-on-loading"
      {% if tab == "test2" %} class="cursor-pointer border-indigo-500 text-indigo-600 dark:text-indigo-500 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" aria-current="page" {% else %} class="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" {% endif %}>
    <span class="text-center sm:text-left">Test 2</span>
  </a>
  <a hx-get="{% url 'route_name' pk=obj.pk %}?search={{ params.items|get_filter_parameters_excluding }}&page=1&order={{ order }}&tab=test3&page_size={{ page_size }}"
      hx-swap="outerHTML"New
      hx-target="#obj-page"
      hx-push-url="true"
      hx-indicator=".transitioned-on-loading"
      {% if tab == "test3" %} class="cursor-pointer border-indigo-500 text-indigo-600 dark:text-indigo-500 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" aria-current="page" {% else %} class="cursor-pointer border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300 group inline-flex items-center py-4 px-1 border-b-2 font-medium text-sm" {% endif %}>
    <span class="text-center sm:text-left">Test 3</span>
  </a>
</nav>
```

with the Loader:
![image](https://github.com/user-attachments/assets/86b237ad-66a9-43d2-876d-4a474b449423)

without the Loader:
![image](https://github.com/user-attachments/assets/91bfbbc4-aaeb-4d25-bce9-3902d83b9799)

Thanks in advance

**Comment by wrabit** [comment]
Hi @pedroven, could you paste me the browser output source code of `<nav>`?

I notice you have `hx-swap="outerHTML"New` but not sure it will be affecting it like this.

**Comment by wrabit** [comment]
@pedroven  Ok I spotted the issue, I will take a look into it, meanwhile, you can fix by keeping the `{% if %}` logic inside the class attribute.

I usually format like this (keeping the constant classes outside of the `{% if %}`

```html
<nav class="-mb-px flex space-x-8" aria-label="Tabs">
    <a hx-get="{% url 'home' %}?search=&page=1&order={{ order }}&tab=test1&page_size={{ page_size }}"
       hx-swap="outerHTML"
       hx-target="#obj-page"
       hx-push-url="true"
       hx-indicator=".transitioned-on-loading"
       class="cursor-pointer py-4 px-1 border-b-2 font-medium text-sm items-center inline-flex group
            {% if tab == 'test1' %}
                border-indigo-500 text-indigo-600 dark:text-indigo-500
            {% else %}
                border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300
            {% endif %}
       "
       aria-current="page">
        <span class="text-center sm:text-left">Test 1</span>
    </a>
...
```

**Comment by wrabit** [comment]
@pedroven I have spotted the issue, I'll take a look into it.

In the meantime it can be fixed by keeping the template expression inside the attribute, which I would normally do anyway. (keeping the consistent classes outside of the `{% if %}`.

```html
<nav class="...">
    <a 
       class="cursor-pointer py-4 px-1 border-b-2 font-medium text-sm items-center inline-flex group
            {% if tab == 'test1' %}
                border-indigo-500 text-indigo-600 dark:text-indigo-500
            {% else %}
                border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300
            {% endif %}
       "
       ...>
        ...
    </a>
...
```

**Comment by pedroven** [comment]
> @pedroven I have spotted the issue, I'll take a look into it.
> 
> In the meantime it can be fixed by keeping the template expression inside the attribute, which I would normally do anyway. (keeping the consistent classes outside of the `{% if %}`.
> 
> ```
> <nav class="...">
>     <a 
>        class="cursor-pointer py-4 px-1 border-b-2 font-medium text-sm items-center inline-flex group
>             {% if tab == 'test1' %}
>                 border-indigo-500 text-indigo-600 dark:text-indigo-500
>             {% else %}
>                 border-transparent text-gray-500 hover:text-gray-700 dark:text-gray-300 dark:hover:text-white hover:border-gray-300
>             {% endif %}
>        "
>        ...>
>         ...
>     </a>
> ...
> ```

It worked! but unfortunately, a similar problem occurs when changing the class value using alpine like this:

```html
<c-icon icon="chevron-down" x-bind:class="open ? 'hidden' : 'mr-2 h-4 w-4'"/>
<c-icon icon="chevron-up" x-bind:class="open ? 'mr-2 h-4 w-4' : 'hidden'"/>
```

**Comment by wrabit** [comment]
@pedroven I've just released 0.9.18 which fixes an issue related to the one-line ternary pattern you used above. Please let me know if this resolves it.

**Comment by wrabit** [comment]
@pedroven the original issue is now resolved in

- #57 

Please let me know if you try and if the issue still exists, thanks!
